app/models/product.rb

```diff
@@ -0,0 +1,56 @@
+class Product < ActiveRecord::Base
+  belongs_to :user
+  has_many :reviews
+  has_many :order_items
+  has_many :product_categories
+  has_many :categories, through: :product_categories
+
+  validates :name, presence: true, uniqueness: true
+  validates :price, presence: true, numericality: { only_integer: true, greater_than: 0 }
+  validates :stock, :user_id, presence: true, numericality: { only_integer: true }
+  validates :user_id, :stock, presence: true
+
+  def avg_rating(reviews)
+    sum = 0
+    count = 0
+    reviews.each do |review|
+      sum += review.rating
+      count += 1
+    end
+    if count == 0
+      0
+    else
+      sum/count
+    end
+  end
+
+  def display_avg(product, reviews)
```

Comment:
Since this is specifically for views this would work really well as a helper method.